v2/pkg/templates/compile.go

```diff
@@ -221,3 +223,55 @@ mainLoop:
 		template.Executer = executer.NewExecuter([]protocols.Request{&offlinehttp.Request{}}, &options)
 	}
 }
+
+// ParseFromReader reads the template from reader
+// returns the parsed template
+func ParseFromReader(reader io.Reader, options protocols.ExecuterOptions) (*Template, error) {
```

Comment:
Not sure if it's intentional, but I'd suggest refactoring the existing `Parse(filePath string, preprocessor Preprocessor, options protocols.ExecuterOptions)` to open the file in `filePath` and pass the reader to `ParseFromReader`. This will also ensure that the Variables block is evaluated through the `Preprocessor`.

Reply:
I agree, we should make `Parse` call `ParseFromReader` so as to reduce duplicate logic